index.js

```diff
@@ -1,29 +1,48 @@
 const passport     = require('passport');
 const ldapStrategy = require('passport-ldapauth');
+const allowedAttributes = [
+  'dn', 'controls', 'objectClass', 'ou', 'o', 'ibmSerialNumber', 'employeeCountryCode', 
+  'isManager', 'managerCountryCode', 'co', 'uid', 'c', 'serialNumber', 'hrActive', 
+  'hrCompanyCode', 'hrCountryCode', 'hrFirstName', 'hrManagerIndicator', 'emailAddress', 
+  'mail', 'notesEmail', 'notesId', 'notesMailDomain', 'notesShortName', 'primaryNode', 'primaryUserId', 
+  'infotelephonenumber', 'personaltitle', 'jobresponsibilities', 'hrEmployeeType', 'employeeType', 
+  'preferredIdentity', 'iptelephonenumber', 'telephoneNumber', 'tieline', 'alternateTelephoneNumber', 
+  'mobile', 'timestampBpgui', 'hrLastName', 'coreDataIntegrity', 'directoryAlias', 'odmEntry', 'div', 
+  'notesMailFile', 'notesMailServer', 'buildingName', 'floor', 'ibmLoc', 'physicalDeliveryOfficeName', 
+  'workLoc', 'workLocation', 'workplaceIndicator', 'givenName', 'sn', 'timestampFeed', 'costCenter', 
+  'dept', 'divDept', 'hrOrganizationCode', 'callupName', 'cn', 'entryType', 'hrOrgUnitCode', 'manager', 
+  'managerSerialNumber', 'timestampHRMS', 'preferredLastName', 'timestampBpguiRI', 'passwordIsExpired', 
+  'passwordisreset', 'passwordIsResetByAdmin', 'passwordIsStruckOut', 'passwordModifyTimestamp']
 
 // Setup passport with IBM's LDAP config
 const config = { server: {
     url: 'ldaps://bluepages.ibm.com:636',
     searchBase: 'ou=bluepages,o=ibm.com',
-    searchFilter: '(mail={{username}})',
-    searchAttributes: ['cn', 'uid', 'mail']
+    searchFilter: '(mail={{username}})'
 }}
-passport.use(new ldapStrategy(config))
-passport.initialize()
 
-module.exports = function(req, res, next) {
+module.exports = {
+  initialize: function(attributes) {
+    if (attributes) {
+      let tmpAttributes = []
+      for (let attribute of attributes) {
+        if (allowedAttributes.indexOf(attribute) > 0) { tmpAttributes.push(attribute) }
+      }
+      config.server.searchAttributes = tmpAttributes
+    }
+
+    passport.use(new ldapStrategy(config))
+    passport.initialize()
+  },
+
+  authenticate: function(req, res, next) {
     passport.authenticate('ldapauth', {session: false}, (err, user, info) => {
-		if (err || !user)
-        	return res.sendStatus(401)
-      	
-      	// Sometimes user and mail can be arrays. Choose the first option
-      	// instead and make it a string
-        if (Array.isArray(user.cn) && user.cn.length > 0)
-			user.cn = user.cn[0]
-        if (Array.isArray(user.mail) && user.mail.length > 0)
-			user.mail = user.mail[0]
+      if (err || !user) { return res.sendStatus(401) }
+      if (Array.isArray(user.cn) && user.cn.length > 0) { user.cn = user.cn[0] }
```

Comment:
hard to do it within the module... I know it is doable but what's the point and the gain from it? I mean, does the guy that is using the API knows how many possibilities are there? for example... I got some cases that people have 3 names in `cn` and others that have 5 names...

Even if we put some sort of safety measures it won't be the same as defining always one position? I mean, if position 5 does not exist return position 0 and also, we will be forcing the guy to do some kind of attempt and error to discover what is the value he wants?

In my humble opinion, I recommend to keep it simple, return the same results as LDAP Strategy, because we are not improving anything here we are just making the life of the guy easier instead to set and configure from the scratch the entire login and interceptor... we don't know what he needs and if we spend some thoughts about it, even him does not know what he needs (it will depend on the project and the needs he is into).

As I said.. if you want me to do it, I will... but for now, I rather keep that way and in the future, we can create a new feature for the module that will allow more customization different results something like that but without taking the possibility of the simple method.